src/Week2/LogAnalysis.hs

```diff
@@ -0,0 +1,133 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module Week1.LogAnalysis where
+
+import Week2.Log
+
+{-
+ Exercise 1 The first step is figuring out how to parse an individual message.
+ Define a function
+    parseMessage :: String -> LogMessage
+ which parses an individual line from the log file.
+
+ For example,
+   parseMessage "E 2 562 help help" == LogMessage (Error 2) 562 "help help"
+-}
+parseMessage :: String -> LogMessage
+parseMessage msg = case words msg of
+  ("I" : timestamp : info)         -> LogMessage Info (read timestamp) (unwords info)
```

Comment:
using read is potentially problematic, because this function is partial. In this case it would fail at runtime if e.g. the 2nd field wasn't an int. Try  something like `parseMessage "E unexpected stuff"`.

In real world app you'd probably use some parsing library, but for the purposes of exercise you could use something like [readMaybe](https://hackage.haskell.org/package/base-4.14.0.0/docs/Text-Read.html#v:readMaybe) and make sure that invalid messages like my example are parsed as `Unknown`

Reply:
You're right. I took it at face value that after "E" an integer must exist.
The current implementation ensures if there's anything unexpected, it gets parsed as an `Unknown` rather than crashing at runtime.
Though I'm not sure if it's really concise to be called readable.

Reply:
Yes, this way it's more robust and you've got a nice idea of also verifying that the numbers are positive.
Readability got somewhat worse and also you're doing some redundant `read` calls (once as a part of readMaybe and once with plain `read`). You could reduce some of that duplication by creating a helper function like:

```haskell
readNat :: String -> Maybe Int
readNat = fmap (\i -> if i >= 0 then Just i else Nothing) . readMaybe
```

There's a big potential to simplify this code much more by using some kind of Parsers & some related type classes (like Functor and Applicative), but that will come in future chapters anyway, so no need to spend too much more time with this exercise :-)